crates/core-new/src/processor.rs

```diff
@@ -0,0 +1,101 @@
+//! Processor is reponsible for running the module.
+//! This run is sound execution of the module.
+//! This will be most abstract layer of the processor.
+
+use std::collections::HashSet;
+
+use anyhow::Result;
+
+use hdp_primitives::task::ExtendedTask;
+use hdp_provider::{
+    evm::{AbstractProvider, AbstractProviderConfig, AbstractProviderResult},
+    key::FetchKeyEnvelope,
+};
+
+use crate::cairo_runner::{
+    input::run::RunnerInput,
+    run::{RunResult, Runner},
+};
+
+pub struct Processor {
+    runner: Runner,
+}
+
+impl Processor {
+    pub fn new() -> Self {
+        let runner = Runner::new();
+        Self { runner }
+    }
+
+    pub async fn process(
+        &self,
+        tasks: Vec<ExtendedTask>,
+        fetch_keys: HashSet<FetchKeyEnvelope>,
+    ) -> Result<RunResult> {
+        // generate input file from fetch points
+        // 1. fetch proofs from provider by using fetch points
+        let config = AbstractProviderConfig {
+            rpc_url: "http://localhost:8080",
+            chain_id: 1,
+            rpc_chunk_size: 1,
+        };
+        let provider = AbstractProvider::new(config);
+        let proofs = provider.fetch_proofs_from_keys(fetch_keys).await?;
+
+        // 2. pre-compute tasks
+
+        // 2. generate input struct with proofs and module bytes
+        let input = self.generate_input(proofs, tasks).await?;
+        // 3. pass the input file to the runner
+        let input_bytes = input.to_bytes();
+        self.runner.run(input_bytes)
+    }
+
+    pub async fn generate_input(
```

Comment:
this is internal function of processor
should we rm pub

Reply:
yup rm and also limit publicity on compilers as it have abstracted compiler